### PR TITLE
Dropdown fix

### PR DESCRIPTION
**Pull request by emplums**
This PR removes an out of place `preventDefault` that was stopping clicks on dropdown items from registering. I am honestly not 100% sure why I felt the need to add this in the first place but I tested out several different scenarios with different dropdowns on the page being open/etc and this should be safe to remove 👍 

Closes: #491 , thanks for the heads up @nicholaai!

### CODE REVIEW

File: pages/components/docs/Dropdown.md [hunk 1]
```diff
@@ -7,7 +7,24 @@ Dropdown.Menu wraps your menu content. Be sure to pass a `direction` prop to thi
 
 ## Default example
 ```.jsx
-<Dropdown title="Dropdown">
+<Dropdown title="Dropdown one">
+  <Dropdown.Menu direction='sw'>
+    <Dropdown.Item><Item 1</Dropdown.Item>
```
Comment (colebemis) [inline pages/components/docs/Dropdown.md]: This stray `<` is breaking the live code example.
```suggestion
    <Dropdown.Item>Item 1</Dropdown.Item>
```

**Comment by emplums** [comment]
@colebemis good catch! I was wondering that too when I was playing around with this today, but thought it was taken care of because that's how the dropdowns in blueprints function, BUT turns out we're creating a totally custom dropdown instead of using this one 😂 I need to spend some time thinking about the right API here

**Comment by emplums** [comment]
@colebemis refreshed a bit on why it's built with a list there - we're expecting users to add links or buttons as children of the `<li>` elements. The lists help with keyboard navigation and is the standard pattern for a11y dropdowns!

**Comment by emplums** [comment]
I think this is ready for a final review now!

**Comment by emplums** [comment]
Ok, so after some rabbit hole adventures, I think this will need to introduce a breaking change. In order to fix the original issue of dropdown menu item clicks being swallowed by the prevent default in the open/close functionality, I needed to introduce a `onToggle` click handler to the wrapper element. Doing this breaks the render prop method because we then would have two toggle tracking functions listening at the same time. 

I'm okay with this change because it seems a little silly to have 3 ways of using this component, and if the user really needs some special functionality to happen `onClick` or `onToggle` for the details element, they could add that behavior in separately.

**Pull request merged**